supler-js/src/CreateFormFromJson.ts

```diff
@@ -5,26 +5,43 @@ module Supler {
     constructor(private renderOptionsGetter:RenderOptionsGetter,
                 private i18n:I18n,
                 private validatorFnFactories:any,
-                private fieldsOptions:FieldsOptions) {
+                private fieldsOptions:FieldsOptions,
+                private fieldOrder: string[][]) {
     }
 
     renderForm(meta,
                formJson,
                formElementDictionary:FormElementDictionary = new FormElementDictionary()):RenderFormResult {
-      var fields = formJson.fields;
-      var fieldCount = fields.length;
+      var fields = formJson.fields.slice();
 
-      var html = this.generateMeta(meta);
+      var rowsHtml = '';
 
-      for (var i = 0; i < fieldCount; i++) {
-        var fieldJson = fields[i];
-        var fieldResult = this.fieldFromJson(fieldJson, formElementDictionary, false);
-        if (fieldResult) {
-          html += fieldResult + '\n';
-        }
+      (this.fieldOrder || formJson.fieldOrder).forEach(row => {
+        rowsHtml += this.row((<string[]>row).map(fieldName => this.findField(fieldName, fields)),
```

Comment:
here it would make sense to revert a change I made recently to have `fields` be not an array, but an object (field name -> field json). That would make the lookups much easier. But we can change this only after duplicate fields are forbidden.